api/api-doc/vector_search.def.json

```diff
@@ -0,0 +1,16 @@
+"vector_search_status": {
+    "id": "vector_search_status",
+    "summary": "Operational status of the Vector Search feature.",
+    "properties": {
+      "status": {
+        "type": "string",
+        "enum": [
```

Comment:
I don't like those custom statuses, but I honestly did not find any standard we can use for such API call. 
(I've looked @ https://learn.microsoft.com/en-us/rest/api/searchservice/http-status-codes and https://docs.couchbase.com/server/current/fts/fts-search-response.html#error-information and neither make a lot of sense to me).